### ISSUE TITLE
Two files with same name in different directories not both compiling

### ISSUE THREAD

**Issue by ashleynolan** [assemble/assemble#747]
I’ve got an issue using v0.6 when I have the following file structure:

- index.hbs
- /process/index.hbs

When compiling, it seems that as these have the same filename, only one is compiled when using the `assemble.src` command:

`assemble.src( 'templates/pages/**/*.hbs' )`

Running this with gulp-debug shows that it is only compiling the second of these files, ignoring the other one.  Any ideas what might cause this?

**Comment by ashleynolan** [comment]
Any idea what this might be?  Haven’t been able to work it out, and means I can’t have root index url’s.

Hoping there must be an easy solution for this…

**Comment by doowb** [comment]
In `assemble@0.6.0-beta.5` you can use `renameKey` to provide a key based on the file path. There's more information [in this issue](https://github.com/assemble/assemble/issues/619).

TLDR;
Set a new `renameKey` option:

```js
// store the original renameKey
var renameKey = assemble.option('renameKey');
// add a new renameKey
assemble.option('renameKey', function (fp) {
  if (fp.indexOf('/index.') !== -1) {
    // get the immediate parent ("process" in your example)
    return path.dirname(fp).split('/').pop();
  }
  // otherwise use the default renameKey function
  return renameKey(fp);
});
```

Hope this helps.

**Comment by ashleynolan** [comment]
That’s great – thanks for that.

**Comment by doowb** [comment]
BTW... i forgot to mention that there will be more granular control over `renameKey` soon.
